applications/launchpad/docker_rig/tarilabs.Dockerfile

```diff
@@ -0,0 +1,194 @@
+# syntax = docker/dockerfile:1.3
+#FROM quay.io/tarilabs/rust_tari-build-with-deps:nightly-2021-11-01 as builder
+
+# Declare TARGETARCH to make it available
+#ARG TARGETARCH
+
+FROM rust:1.60-bullseye as builder
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETVARIANT
+
+#RUN apt update && apt-get --no-install-recommends install -y \
+
+# Disable anti-cache
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+# https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/syntax.md#run---mounttypecache
+RUN --mount=type=cache,id=build-apt-cache-${TARGETOS}-${TARGETARCH}${TARGETVARIANT},sharing=locked,target=/var/cache/apt \
+    --mount=type=cache,id=build-apt-lib-${TARGETOS}-${TARGETARCH}${TARGETVARIANT},sharing=locked,target=/var/lib/apt \
+  apt update && apt-get install -y \
+  apt-transport-https \
+  bash \
+  ca-certificates \
+  curl \
+  gpg \
+  iputils-ping \
+  less \
+  libreadline-dev \
+  libsqlite3-0 \
+  openssl \
+  telnet \
+  cargo \
+  clang \
+  cmake
+
+WORKDIR /tari
+
+# Adding only necessary things up front and copying the entrypoint script last
+# to take advantage of layer caching in docker
+ADD Cargo.lock applications/deps_only/Cargo.lock
+ADD rust-toolchain.toml .
+
+#RUN pwd
+#RUN ls
+#RUN ls /
+#RUN ls /tari
+
+ARG ARCH=native
+#ARG FEATURES=avx2
+ARG FEATURES=safe
+ENV RUSTFLAGS="-C target_cpu=$ARCH"
+ENV ROARING_ARCH=$ARCH
+ENV CARGO_HTTP_MULTIPLEXING=false
+
+ARG APP_NAME=wallet
```

Comment:
Maybe we should make the default something like "MISSING_APP_NAME" so that the build will fail / clearly not work if it's not provided.
Otherwise, we may accidentally build "wallet" when we should be building something else.

Similarly for other app-specific defaults.

Reply:
My thinking is we should have a good default, ie build all apps in one image (would be very long compile), or build quickest binary or tool or the most useful binary - wallet? Would be nice to have a successful build and the find you need to add some settings for other tools.

 Not sure if we can do a MISSING_APP_SETUP that would not seem broken if somebody has not read the docs. Let me see what we can come up with.